### PR TITLE
feat(torch): torch.histc function added to torch front-end

### LINKED ISSUE #23527: histc

### PR DESCRIPTION
**Pull request by kiluazen**
# PR Description

## Related Issue

Close #23527

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [ ] Did pre-commit not fail on any check?
- [ ] Did you follow the steps we provided?

### Socials:

@KushalSM5

### CODE REVIEW

**Comment by kiluazen** [comment]
My tests failed, but had to make the PR, else my issue was gonna expire. I need assistance in understanding the errors.

**Comment by kiluazen** [comment]
I have solved results not matching error for numpy vs torch ground truth. Essentially the problem is when min and max of the array are equal then how are you gonna make bins for the histogram.
I used (min - 5.0000e-1, max +5.0000e+1) that's how they do it in torch. 
But there is a fundamental difference in the implementations of tensorflow and torch I think, which I couldn't resolve.
Lastly,
dtypes not matching. I needed assistance on what should I do about that.
Thank you.

**Comment by jieunboy0516** [comment]
hi @kiluazen , i really apologise for my late reply. It seems that there are falsifying examples shown in the unit tests. Could you take a look and fix it please?
image
thanks a lot for your contribution to ivy!

**Comment by kiluazen** [comment]
I am struggling with understanding the root cause of my test failures. please take a look at this colab [file](https://colab.research.google.com/drive/1p3i6s2Ta03Zs3Ug6K4uhgI8Br73tEL-D?usp=sharing). I played with ouputs of jax.numpy and torch.histc and also the histc funciton I wrote using ivy.histogram. They all are giving me the same outputs. But in my failed test cases it's like backend jax is not matching with ground truth torch. So maybe there is conversion problem with frontend to jax backend, I don't know . 

I need some assistance with this. Can you specify a 25 min window in the next week, so we can both be online and have a back and forth conversation and resolve this issue.

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi @kiluazen 
Thanks for the PR. Sorry for the delay on our end.
Could you remove the duplicate histc implmentations and the tests and cleanup stuff from conflict resolving like  `<<<<<<< HEAD` and `=======`
Feel free to request a review when done. Thanks :)

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
